### ISSUE TITLE
Masked BERT model from keras_nlp produces weird results

### ISSUE THREAD

**Issue by vctrs** [keras-team/keras-hub#1963]
**Description**

I am trying to use keras-nlp with a pretrained masked BERT model to predict some tokens in a sequence. However the model produces inconsistent results. Am I doing something wrong?

**To Reproduce**

https://colab.research.google.com/drive/1kL5oXveRen9Zub_7kJCI28Gg34GfDt_x

**Expected behavior**

Tokens in predictions should match tokens in labels. The preprocessor masks two tokens, and first two tokens of predictions probably should match labels.

**Comment by mehtamansi29** [comment]
Hi @vctrs -

Thanks for reporting the issue. I have tested the code snippet and reproduces the reported behaviour. 
Attached [gist](https://colab.sandbox.google.com/gist/mehtamansi29/0676cb941f9375c6e62fdf328c9a70ad/1963-masked-bert-model-from-keras_nlp-produces-weird-results.ipynb) file for reference.

We will look into the issue and update you the same.

**Comment by mrinalghoshh** [comment]
The issue still persist in keras_hub. Attached[ gist](https://colab.research.google.com/gist/mrinalghoshh/74698d089437170bf1e674c67fa8c9c0/1963-masked-bert-model-from-keras_nlp-produces-weird-results.ipynb) for reference

**Comment by maitry63** [comment]
Hi @vctrs , Thanks for Reporting this! 

I want through this & with latest KerasHub , got the same  output. 

I think the `LABELS` had mostly `0` because the preprocessor randomly masked only a few words and set all other positions to `0` and the `PREDICTIONS` had mostly `26596` because most of the input sequence was padding. The predicted words looked random because the MLM prediction head is not seeems to be pretrained.

Tried with an uncased model and a `[MASK]` token, and predicted only the masked position. However, the output was still a random subword like `##oping`.

Check this [gist](https://colab.sandbox.google.com/gist/maitry63/29962634f0da49f9229a4d26d717962c/1963-masked-bert-model-from-keras_nlp-produces-weird-results.ipynb#scrollTo=ephX8LExYRyo).

I think, this happens because KerasHub loads only the pretrained BERT backbone. The MLM prediction head is randomly initialized and is not loaded from the preset. 

According to this [documentation](https://keras.io/keras_hub/api/models/bert/bert_masked_lm/) , `BertMaskedLM` is meant to be trained first, and only the backbone weights are loaded from the preset.  So, random predictions like `##oping` are expected until the MLM head is trained. It does not seems to be a bug!
